### DISCUSSION TITLE
CreateConsumerAsync without creating queue

### DISCUSSION THREAD

**Discussion by pgarbo** [Havret/dotnet-activemq-artemis-client#408] [Q&A]
Hi,

Is there a way to call CreateConsumerAsync without creating an address/queue if it does not exist? The problem we have is that server creates a very specific type of queue dynamically and we don't want clients to create it with default parameters before server does it.

**Comment by Havret** [comment]
Hello @pgarbo,

Yes, you need to use factory method that accepts address and queue name:

```csharp
var endpoint = Endpoint.Create("localhost", 5672, "guest", "guest");
var connectionFactory = new ConnectionFactory();
var connection = await connectionFactory.CreateAsync(endpoint);
var consumer = await connection.CreateConsumerAsync(address: "my-address", queue: "my-queue");
```

If the queue doesn't exist, `CreateConsumerAsync` will throw an exception. 

I hope that helps,
Havret

**Reply by pgarbo** [reply]
Hi, 

The code example does not throw exception, it is just creating a multicast queue for me:
![image](https://user-images.githubusercontent.com/16103019/208068068-7c9b99e9-13e1-497b-9b98-4bcb0aefa67c.png)

I am using version 2.11.0.

Pawel

**Reply by Havret** [reply]
I see, you probably have `auto-create-queues` setting set to true on the broker. It mainly working with this setting set to false.

The only other thing I can think of is using topology manager to check if queue with the given name already exists. Or you could restrict permissions to `createDurableQueue` only to your server application.
